login-utils/login.c

```diff
@@ -1287,6 +1288,39 @@ static void __attribute__((__noreturn__)) usage(void)
 	exit(EXIT_SUCCESS);
 }
 
+static void load_credentials(struct login_context *cxt) {
+	char *env;
+	DIR *dir;
+	struct dirent *d;
+	struct path_cxt *pc;
+
+	env = safe_getenv("CREDENTIALS_DIRECTORY");
+        if (!env)
+                return;
+
+	pc = ul_new_path("%s", env);
+	if (!pc) {
+		syslog(LOG_ERR, _("failed to initialize path context"));
+		exit(EXIT_FAILURE);
+	}
+
+	dir = ul_path_opendir(pc, NULL);
+	if (!dir) {
+		syslog(LOG_ERR, _("failed to open credentials directory"));
+		exit(EXIT_FAILURE);
+	}
```

Comment:
I don't think exit() is a good reaction when I see the patch again. 

It is login(1); if you lost the directory (or forgot to create it or define the env. variable in a bad way), you will not be able to log in to the system. That's a serious problem for admins. It would be better to return from the function and continue without credentials.

I think
```
    syslog(LOG_WARNING, _("failed to initialize path context"));
    return;
```
would be better for both cases.